### PR TITLE
Chapter_9

### PR DESCRIPTION
**Pull request by hpt1512**
1. Link excel: https://docs.google.com/spreadsheets/d/1EYHDUr9EZArzCBvuTu1KBFbGnDhHCCfbDMHw0CzBH8g/edit?usp=sharing
2. Rubocop
image

### CODE REVIEW

**Comment by hpt1512** [comment]
Ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -6,6 +6,15 @@ def log_in user
   # Returns the current logged-in user (if any).
   def current_user
     @current_user ||= User.find_by id: session[:user_id]
+    if (user_id = session[:user_id])
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]:
```suggestion
    if user_id = session[:user_id]
```
hạn chế dùng ngoặc. chỗ nào bỏ dc thì bỏ nhé e

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -14,7 +23,22 @@ def logged_in?
   end
 
   def log_out
+    forget(current_user)
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]:
```suggestion
    forget current_user
```

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -14,7 +23,22 @@ def logged_in?
...
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  # Forgets a persistent session.
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]:
```suggestion
    cookies.delete :user_id
```

**Comment by hpt1512** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**